### PR TITLE
add audit log for account lock

### PR DESCRIPTION
**Pull request by Sathiyakugan**
Fixes wso2/product-is#9119

### CODE REVIEW

**Review by tharindu-bandara — APPROVED** [review]

File: components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java [hunk 1]
```diff
@@ -734,4 +741,40 @@ private void setUserClaim(String claimName, String claimValue, UserStoreManager
             throw new AccountLockException("Error while setting user claim value :" + username, e);
         }
     }
+
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java:744]: Additional space

File: components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java [hunk 2]
```diff
@@ -734,4 +741,40 @@ private void setUserClaim(String claimName, String claimValue, UserStoreManager
...
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = AuditConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        loggedInUser = UserCoreUtil.addTenantDomainToEntry(loggedInUser, tenantDomain);
+        AUDIT_LOG.info(String.format(AuditConstants.AUDIT_MESSAGE, loggedInUser, action, target, dataObject, result));
+    }
+
+    private void auditAccountLock(String action, String target, String userStoreDomainName, boolean isAdminInitiated,
```
Comment (ivantha) [inline components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java:765]: Not strictly necessary but it is better if you can add a doc comment for this method stating the functionality.

File: components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java (hunk 1 above)
Comment (omindu) [reply components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java:744]: Fixed with https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/73

File: components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java (hunk 2 above)
Comment (omindu) [reply components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java:765]: Fixed with https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/73

**Pull request merged**